integration-tests/xml/pom.xml

```diff
@@ -35,14 +35,30 @@
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-xslt</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-xslt-saxon</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-xpath</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-bean</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-direct</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-file</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-mock</artifactId>
+        </dependency>
```

Comment:
@zhfeng I have some concerns about adding this many new dependencies to the test. I mean it is generally better to test with as little dependencies as possible, so that it does not happen that an extension that is there just for testing some niche feature brings some build steps that become vital for the execution of some basic functionality.

Wouldn't it perhaps be worthwhile to introduce a new XML testing group like we already have with [Azure, AWS an others](https://github.com/apache/camel-quarkus/tree/main/integration-test-groups)? We could have several XML modules there that would test the use cases coverable with their minimal set of deps and ideally still be able to run them all together on the CI (in the grouped module), so that the exec time does not explode. See also https://camel.apache.org/camel-quarkus/2.11.x/contributor-guide/extension-testing.html#_grouping
WDYT?

Reply:
> But how to group the sub modules?

The grouping is given by the required dependencies. I.e. if the `bean:` test protocol requires an additional `camel-quarkus-bean` dependency, then it should be a separate test module with that dependency under the new group.

Reply:
OK, let's raise a follow up issue to group the xml tests.